hack/rules/redhat_ci.md

```diff
@@ -0,0 +1,65 @@
+# redhat-ci
+
+This rule checks for compliance with some coding conventions in the OCP
+collection
+
+The `redhat-ci` rule has the following checks:
+
+- `redhat-ci[no-role-prefix]` - Variables used within a role should have a
+  prefix related to the role. This includes:
+  1. Variables passed to `{include,import}_role` and `{include_import}_tasks`
+     should be named `$prefix_$var`
+  1. Facts saved from within roles should be named `$prefix_$fact`
+  1. Variables registered from tasks to use in subsequent tasks should be named
+     `_$prefix_$var` - notice the extra underscore to signal this variable is
+     "private"
+
+  The rules to figure out the prefix are:
+  1. If the role name is $SHORT (where $SHORT is < 6 characters) use the whole
+     name
+  1. If not, split the role name by underscores into words
+  1. If there's a single word, use the first $SHORT characters
+  1. Finally, if there are 2 or more words, build an acronym from the split
+     words
+
+  Examples:
+  | Role Name               | Prefix    |
+  |-------------------------|-----------|
+  | `sno`                   | `sno_`    |
+  | `installer`             | `insta_`  |
+  | `validate_http_store`   | `vhs_`    |
```

Comment:
Would it be possible for this custom profile to include the var-naming role-prefix, just in case any role is already following that? otherwise it will need to switch back to use these rules.
```Shell
  | Role Name       | Prefix           |
  |-----------------|------------------|
  | custom_role_foo | custom_role_foo_ |
```